extensions/AbuseFilter/AbuseFilterVariableHolder.php

```diff
@@ -411,4 +397,19 @@ function compute( $vars ) {
 		return $result instanceof AFPData
 			? $result : AFPData::newFromPHPVar( $result );
 	}
+
+	private function getRecentAuthorsNames( $articleId, $cutOff ): array {
+		$dbr = wfGetDB( DB_SLAVE );
+		$res = $dbr->select( 'revision', 'DISTINCT rev_user, rev_user_text', [
+			'rev_page' => $articleId,
+			'rev_timestamp<' . $dbr->addQuotes( $dbr->timestamp( $cutOff ) ),
+		], __METHOD__, [ 'ORDER BY' => 'rev_timestamp DESC', 'LIMIT' => 10 ] );
+
+		$users = [];
+		foreach ( $res as $row ) {
```

Comment:
We can refine it to use `User::whoAre()` which will reduce the number of queries in favour of an additional PHP iteration.  Considering the `LIMIT = 10`, 10 times less queries in favour of a quick iteration over 10-element long array.